### PR TITLE
Adds a cancel button to the drop bomb command.

### PR DESCRIPTION
**Pull request by Spacemanspark**
why is there no option to cancel out right now

### CODE REVIEW

**Comment by Lady-Luck** [comment]
A lot of prompts lack a cancel button. Like doing a station announcement from the bridge.

**Comment by FlattestGuitar** [comment]
That's still gonna make the logs exist. You need to add an option to the switch statement that returns the thing.

💥

**Comment by Crazylemon64** [comment]
blow it all to hell

**Comment by Spacemanspark** [comment]
@Lady-Luck if you press "ok" with no text it basically automatically cancels. With a bomb you really can't do that.

**Comment by Lady-Luck** [comment]
>if you press "ok" with no text in the box, it (basically) automatically cancels. With the bomb command you really can't do that.

it would be so much less of a big deal if ctrl+a actually worked.

**Comment by Spacemanspark** [comment]
Ctrl + shift + left arrow key.

**Comment by Lady-Luck** [comment]
I know about that hotkey, yes. But for some reason when you use it in BYOND it has a tendency to leave a trailing space, I've accidentally sent just a single space as a station announcement before.

Plus not a lot of people know about that hotkey.

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -552,7 +552,7 @@ var/list/admin_verbs_snpc = list(
 		return
 
 	var/turf/epicenter = mob.loc
-	var/list/choices = list("Small Bomb", "Medium Bomb", "Big Bomb", "Custom Bomb")
+	var/list/choices = list("Small Bomb", "Medium Bomb", "Big Bomb", "Custom Bomb", "Cancel")
 	var/choice = input("What size explosion would you like to produce?") in choices
```
Comment (Fox-McCloud) [inline code/modules/admin/admin_verbs.dm]: Not needed

`var/choice = input("What size explosion would you like to produce?") as null|anything in choices`

**Comment by Spacemanspark** [comment]
Fox, remind me tomorrow, discord exploded and I'm at work

**Comment by Allfd** [comment]
@Spacemanspark 

This is bubble wrapping!

**Comment by Fox-McCloud** [comment]
@Spacemanspark

**Comment by Spacemanspark** [comment]
oh shit whoops one moment

**Comment by Spacemanspark** [comment]
@Fox-McCloud glance at discord when you get the chance

**Comment by Fox-McCloud** [comment]
I know you updated this; ever get around to testing it?

**Comment by Spacemanspark** [comment]
Yessssss, one moment.

File: code/modules/admin/admin_verbs.dm [hunk 2]
```diff
@@ -552,8 +552,8 @@ var/list/admin_verbs_snpc = list(
 		return
 
 	var/turf/epicenter = mob.loc
-	var/list/choices = list("Small Bomb", "Medium Bomb", "Big Bomb", "Custom Bomb")
-	var/choice = input("What size explosion would you like to produce?") in choices
+	var/list/choices = list("Small Bomb", "Medium Bomb", "Big Bomb", "Custom Bomb", "Cancel")
```
Comment (Fox-McCloud) [inline code/modules/admin/admin_verbs.dm]: You can remove "Cancel" now, or should be able to.

File: code/modules/admin/admin_verbs.dm [hunk 3]
```diff
@@ -552,8 +552,8 @@ var/list/admin_verbs_snpc = list(
 		return
 
 	var/turf/epicenter = mob.loc
-	var/list/choices = list("Small Bomb", "Medium Bomb", "Big Bomb", "Custom Bomb")
-	var/choice = input("What size explosion would you like to produce?") in choices
+	var/list/choices = list("Small Bomb", "Medium Bomb", "Big Bomb", "Custom Bomb", "Cancel")
+	var/choice = input("What size explosion would you like to produce?") as null|anything in choices
 	switch(choice)
 		if(null)
 			return 0
```
Comment (Fox-McCloud) [inline code/modules/admin/admin_verbs.dm]: I'm not sure if `if(null) return 0` needs to be here.

Comment (Spacemanspark) [inline code/modules/admin/admin_verbs.dm]: I didn't test it without, so I dunno.

File: code/modules/admin/admin_verbs.dm (hunk 2 above)
Comment (Spacemanspark) [inline code/modules/admin/admin_verbs.dm]: Wait I thought I removed that.
Whoops.

**Pull request merged**